In [1]:
import pandas as pd
import numpy as np

In [8]:
rating_df = pd.read_csv( "ml-100K/ml-100k/u.data", delimiter = "\t", header = None )
rating_df.columns = ["userid", "movieid", "rating", "timestamp"]

In [3]:
rating_df.head(10)

userid  movieid  rating  timestamp
0     196      242       3  881250949
1     186      302       3  891717742
2      22      377       1  878887116
3     244       51       2  880606923
4     166      346       1  886397596
5     298      474       4  884182806
6     115      265       2  881171488
7     253      465       5  891628467
8     305      451       3  886324817
9       6       86       3  883603013

In [9]:
#Number of unique users
len( rating_df.userid.unique() )

943

In [10]:
#Number of unique movies
len( rating_df.movieid.unique() )

1682

### So a total of 1682 movies and 943 users data is available in the dataset.

In [11]:
#Let's drop the timestamp columns. We do not need it.
rating_df.drop( "timestamp", inplace = True, axis = 1 )
rating_df.head( 10 )

userid  movieid  rating
0     196      242       3
1     186      302       3
2      22      377       1
3     244       51       2
4     166      346       1
5     298      474       4
6     115      265       2
7     253      465       5
8     305      451       3
9       6       86       3

### Load Movies Data

In [13]:
movies_df = pd.read_csv( "ml-100k/ml-100k/u.item", delimiter = '\|', header = None )

C:\Users\Sachin.Manjhi\AppData\Local\Continuum\anaconda3\lib\site-packages\ipykernel_launcher.py:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.
  """Entry point for launching an IPython kernel.


In [17]:
movies_df.head( 10 )

movieid                                              title
0        1                                   Toy Story (1995)
1        2                                   GoldenEye (1995)
2        3                                  Four Rooms (1995)
3        4                                  Get Shorty (1995)
4        5                                     Copycat (1995)
5        6  Shanghai Triad (Yao a yao yao dao waipo qiao) ...
6        7                              Twelve Monkeys (1995)
7        8                                        Babe (1995)
8        9                            Dead Man Walking (1995)
9       10                                 Richard III (1995)

# Finding User Similarities

In [18]:
from sklearn.metrics import pairwise_distances
from scipy.spatial.distance import cosine, correlation

In [19]:
# Create the pivot table
user_movies_df = rating_df.pivot( index='userid', columns='movieid', values = "rating" ).reset_index(drop=True)

In [20]:
#Fill '0' for ratings not given by users
user_movies_df.fillna( 0, inplace = True )

In [21]:
#Let look into the movie rating data
user_movies_df.iloc[10:20, 20:30]

movieid   21   22   23   24   25   26   27   28   29   30
10       0.0  4.0  0.0  3.0  3.0  0.0  0.0  5.0  3.0  0.0
11       0.0  0.0  0.0  0.0  0.0  0.0  0.0  5.0  0.0  0.0
12       3.0  4.0  5.0  1.0  1.0  0.0  3.0  5.0  2.0  0.0
13       0.0  3.0  5.0  0.0  2.0  0.0  0.0  0.0  0.0  0.0
14       0.0  0.0  0.0  0.0  3.0  0.0  0.0  0.0  0.0  0.0
15       0.0  5.0  0.0  0.0  0.0  0.0  2.0  5.0  0.0  0.0
16       0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
17       0.0  5.0  4.0  0.0  3.0  4.0  0.0  3.0  0.0  0.0
18       0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0
19       0.0  5.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0  0.0